### PR TITLE
explicitly add IConfigurationListener to TestRunner

### PR DESCRIPTION
**Pull request by missedone**
fixed https://github.com/cbeust/testng-eclipse/issues/298

this is a workaround that fix at testng-remote side, in essentially, need properly fixed at testng.

ping @cbeust , @juherr

### CODE REVIEW

File: remote6_9_10/src/main/java/org/testng/remote/support/RemoteTestNG6_9_10.java [hunk 1]
```diff
@@ -32,6 +33,9 @@ public TestRunner newTestRunner(ISuite suite, XmlTest xmlTest,
               runner.addListener(new TestHTMLReporter());
               runner.addListener(new JUnitXMLReporter());
             }
+            for (IConfigurationListener cl : getConfiguration().getConfigurationListeners()) {
```
Comment (missedone) [inline remote6_9_10/src/main/java/org/testng/remote/support/RemoteTestNG6_9_10.java]: this init should be in org.testng.TestRunner.init():

```
  private void init() {
    initMetaGroups(m_xmlTest);
    initRunInfo(m_xmlTest);

    // Init methods and class map
    // JUnit behavior is different and doesn't need this initialization step
    if(!m_xmlTest.isJUnit()) {
      initMethods();
    }

    initListeners();
    addConfigurationListener(m_confListener);
// Nick: add configuration listener here.
    for (IConfigurationListener cl : m_configuration.getConfigurationListeners()) {
      addConfigurationListener(cl);
    }
  }
```

Comment (missedone) [inline remote6_9_10/src/main/java/org/testng/remote/support/RemoteTestNG6_9_10.java]: actually, might need to refactor testng to uniform addConfigurationListener at a single source or so, for now there's too many places missed calling this.

**Comment by juherr** [comment]
LGTM :+1:

Once merged and released in testng itself, we will have to check if the listeners are no registered twice. I think we will have to add a new remote impl without the fix.

**Comment by juherr** [comment]
LGTM

(I'm playing with lgtm.co and I configured it)

**Comment by missedone** [comment]
I check with the both new of testng-remote and testng, the configuration listener won't be registered twice, since internally a HashSet is used to store them, and works so far even does not override `hashCode` and `equals`. and I agree that it'll be safe to have a new factory for new testng.

**Comment by juherr** [comment]
We can trust users ;p

**Pull request merged**